examples/dataflow-python-examples/dataflow_python_examples/streaming/sideinput_refresh.py

```diff
@@ -0,0 +1,450 @@
+                                 k=len(categories))  # Shuffles data every time
+
+    def process(self,
+                element,
+                timestamp=beam.DoFn.TimestampParam,
+                window=beam.DoFn.WindowParam,
+                pane_info=beam.DoFn.PaneInfoParam):
+        """Called once for each side input type whenever the side input data needs to be refreshed
+           Fetches look up values, associates the look up values to the corresponding products
+
+         Args:
+             element(str): Pipeline input element
+             timestamp: Timestamp associated with an input element assigned by source
+             window: window information
+             pane_info: Pane details
+
+         Returns:
+            Tuple containing product name and corresponding look up value
+         """
+        lookup_values = self._get_lookup_values()
+
+        window_start = get_formatted_time(window.start, time_zone='UTC')
+        window_end = get_formatted_time(window.end, time_zone='UTC')
+        logging.info(
+            f"Timestamp:{timestamp};Window Start{window_start}; Window end:{window_end}; Pane Info:{pane_info}",
+            self.sideinput_type)
+
+        products_lookupvalues = [(product, lookup_values[index])
+                                 for index, product in enumerate(self.products)]
+        logging.info(
+            ",".join([
+                f"{productinfo[0]} - {productinfo[1]}"
+                for productinfo in products_lookupvalues
+            ]), self.sideinput_type)
+
+        for product in products_lookupvalues:
+            yield product
+
+
+@beam.typehints.with_input_types(bytes)
+@beam.typehints.with_output_types(beam.pvalue.TaggedOutput)
+class SplitToMultiple(beam.DoFn):
+    """Generates a base path for each side input type combining root path received via file notification subscription
+       and side input type
+
+       Attributes:
+         sideinput_types: List of Side input types
+         file_prefix: file_prefix matching required files. Default is * indicating all files
+    """
+
+    def __init__(self, sideinput_types: List[str], file_prefix: str = "*"):
+        self.sideinput_types = sideinput_types
+        self.file_prefix = file_prefix
+
+    def process(self,
+                element,
+                timestamp=beam.DoFn.TimestampParam,
+                window=beam.DoFn.WindowParam,
+                pane_info=beam.DoFn.PaneInfoParam):
+        logging.info(
+            f"(Re)loading side input data for window: {get_formatted_time(window.start)} - {get_formatted_time(window.end)}"
+        )
+        for sideinput_type in self.sideinput_types:
+            yield beam.pvalue.TaggedOutput(
+                sideinput_type,
+                FileSystems.join(element.decode(), sideinput_type,
+                                 self.file_prefix))
+
+
+@beam.typehints.with_input_types(Tuple[str, Iterable[Dict[str, Any]]])
+class PrintEvents(beam.DoFn):
```

Comment:
Could we sink to pubsub and have a little cli utility that tails the pubsub topic in your shell?
Or just sink to BQ and loop a select * query?

Reply:
This would allow you to have a powerful demo script that tails the enriched pubsub for a few seconds with initial side input values and then logs 
```
********** UPLOADING NEW SIDE INPUT DATA *************
```
and continues to tail pubsub showing the updated enriched data.